ag_data/formulas/ingestion_engine.py

```diff
@@ -0,0 +1,33 @@
+from ag_data import models
+from ag_data.formulas.library.system import mercury_formulas as hgFormulas
+
+
+class MeasurementIngestionEngine:
+
+    processingFormulas = {
+        0: hgFormulas.fPass,
+        2: hgFormulas.fMercurySimpleTemperatureSensor,
+        4: hgFormulas.fMercuryDualTemperatureSensor,
+        6: hgFormulas.fMercuryFlowSensor,
+    }
+
+    def saveMeasurement(self, measurementDict, event):
+        timestamp = measurementDict["measurement_timestamp"]
+        sensor_id = measurementDict["measurement_sensor"]
+        rawValue = measurementDict["measurement_values"]
```

Comment:
If someone writes a code calling `saveMeasurement` function, they must know which keys to put inside into the dictionary. And there's no way to figure out what to include in the dict other than reading the actual implementation.

Also the implementation assumes all those 3 fields exist. To me, it's more like function parameters.
So my suggestion would be one of the followings:

1. At least give it a documentation of what should be inside of the dictionary. It doesn't have to be verbose but listing three mandatory keys with a short description would be enough.
2. Replace measurementDict with three separate function parameters.
3. Use a dedicated data class instead of dictionary.

Personally, I prefer the last option.

Also consider removing unnecessary `measurement` prefix. it's already a key of "measurement"Dict and I believe everyone can easily infer it's not just a timestamp but a timestamp for a measurement.

Reply:
The reason here is a dictionary is because, this engine was designed to take in what we receive from the hardware-facing measurement receiver API. I agree it is not robust, readable or elegant. What I plan do to fix this, is to create a new class that encloses these three things, and pass them around.

Reply:
So to have a dedicated data class, what about using a newly created but unsaved AGMeasurement record?

Reply:
I don't think it would be a good idea of using a django-backend model (without saving it).

I guess python 3.7 is not requirement of this project so we cannot use [dataclasses](https://docs.python.org/3.7/library/dataclasses.html).

How about creating it like:
```
class ABC:
    def __init__(self, timestamp, ...):
        self.timestamp =...
```

Or you can try [attr.s](http://www.attrs.org/en/stable/examples.html). You can take a look at [its use cases from the last semester](https://github.com/gcivil-nyu-org/fall2019-cs-gy-6063-team-five_fries/blob/develop/external/models.py#L40)

Reply:
I am moving this to #377 for further design discussion and code changes.